render_block/django.py

```diff
@@ -61,20 +68,31 @@ def _build_block_context(template, context):
     block_context = context.render_context[BLOCK_CONTEXT_KEY]
 
     for node in template.nodelist:
+
         if isinstance(node, ExtendsNode):
+            # Add the block nodes from this node to the block context
+            block_context.add_blocks(node.blocks)
+
             compiled_parent = node.get_parent(context)
 
             # Add the parent node's blocks to the context. (This ends up being
             # similar logic to ExtendsNode.render(), where we're adding the
             # parent's blocks to the context so a child can find them.)
-            block_context.add_blocks(
-                {
-                    n.name: n
-                    for n in compiled_parent.nodelist.get_nodes_by_type(BlockNode)
-                }
-            )
-
-            _build_block_context(compiled_parent, context)
+            for sub_node in compiled_parent.nodelist:
```

Comment:
This change is to add any parent nodes to the context *except* the extends node?